pouchdb-tests.ts

```diff
@@ -374,6 +374,28 @@ module PouchDBTest {
                 db2.put(bDoc, id, {}).then(resp => { }, err => { }).catch(err => { });
             }
         }
+
+        module query {
+
+            function callback() {
+                //var db: pouchdb.callback.PouchDB = new PouchDB("dbname", (e, v) => { });
+
+            }
+
+            function promise() {
+                var db: pouchdb.thenable.PouchDB = new PouchDB("dbname");
+                var qr: pouchdb.async.PouchPromise<pouchdb.api.methods.allDocs.Response>;
+                qr = db.query('testview');
+                qr = db.query((doc, emit) => { emit(doc.name); });
+                qr = db.query({
+                    map: (doc, emit) => { emit(doc.name); },

```

Comment:
Although at first glance having `emit` faked as a second parameter looks like a nice way to go around [this issue](https://github.com/Microsoft/TypeScript/issues/5966#issuecomment-162629962), which is bloking full typed support for the query api, `(doc, emit) => { emit(doc.name); }` is not a valid code for the map function:

- Unfortunately, `emit` is not passed as a second parameter when used as full map/reduce object or view reference. See [here](http://pouchdb.com/api.html#closures). This approach would only make sense when used like in line 389.